node-graph/gcore/src/graphic_element/renderer.rs

```diff
@@ -423,10 +424,26 @@ impl GraphicElementRendered for VectorData {
 				Some(color) => peniko::Color::rgba(color.r() as f64, color.g() as f64, color.b() as f64, color.a() as f64),
 				None => peniko::Color::TRANSPARENT,
 			};
+			use crate::vector::style::{LineCap, LineJoin};
+			use vello::kurbo::{Cap, Join};
+			let cap = match stroke.line_cap {
+				LineCap::Butt => Cap::Butt,
+				LineCap::Round => Cap::Round,
+				LineCap::Square => Cap::Square,
+			};
+			let join = match stroke.line_join {
+				LineJoin::Miter => Join::Miter,
+				LineJoin::Bevel => Join::Bevel,
+				LineJoin::Round => Join::Round,
+			};
 			let stroke = kurbo::Stroke {
 				width: stroke.weight,
 				miter_limit: stroke.line_join_miter_limit,
-				..Default::default()
+				join,
+				start_cap: cap,
+				end_cap: cap,
+				dash_pattern: stroke.dash_lengths.into(),
+				dash_offset: stroke.dash_offset,
```

Comment:
There's a mismatch in behavior with how dash_offset affects dash_pattern: 

https://github.com/user-attachments/assets/220484a6-e428-402f-9f0e-7e98f8c4c64f

Please also change the properties layout code such that:
- You fix a bug I just found with `dash_offset` where negative values should be valid, but currently it's getting clamped to 0 when you try to type a negative value.
- Dash Offset is grayed out (`.disabled(true)` is set on its widget builder pattern) when Dash Lengths is empty.
- Miter Limit is grayed out when Line Join isn't set to Miter.

Reply:
It might be that this is an upstream `vello` issue, I don't see how my code would break this in any way

Reply:
Added the properties panel improvements

Reply:
If you think that's a Vello issue, can you please ask them about it?

Reply:
https://xi.zulipchat.com/#narrow/stream/197075-gpu/topic/Dashed.20Line.20misbehaves.20when.20used.20with.20dash-offset/near/455494308